assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -39,45 +44,69 @@ import { numFmt } from '../../util';
 const { useSelect } = Data;
 
 const WPDashboardPopularPages = ( { WidgetReportZero, WidgetReportError } ) => {
-	const dateRangeDates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
+	const { report, titles, loading, error } = useSelect( ( select ) => {
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( {
 			compare: true,
 			offsetDays: DATE_RANGE_OFFSET,
-		} )
-	);
+		} );
+		const reportArgs = {
+			...dateRangeDates,
+			metrics: [
+				{
+					expression: 'ga:pageviews',
+					alias: 'Pageviews',
+				},
+			],
+			dimensions: [ 'ga:pagePath' ],
+			orderby: [
+				{
+					fieldName: 'ga:pageviews',
+					sortOrder: 'DESCENDING',
+				},
+			],
+			limit: 5,
+		};
+		const data = {
+			report: select( MODULES_ANALYTICS ).getReport( reportArgs ),
+			error: select( MODULES_ANALYTICS ).getErrorForSelector(
+				'getReport',
+				[ reportArgs ]
+			),
+			loading: true,
+		};
+		const reportLoaded = select(
+			MODULES_ANALYTICS
+		).hasFinishedResolution( 'getReport', [ reportArgs ] );
 
-	const reportArgs = {
-		...dateRangeDates,
-		metrics: [
-			{
-				expression: 'ga:pageviews',
-				alias: 'Pageviews',
-			},
-		],
-		dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
-		orderby: [
-			{
-				fieldName: 'ga:pageviews',
-				sortOrder: 'DESCENDING',
-			},
-		],
-		limit: 5,
-	};
+		let pagePaths = [];
+		let hasLoadedPageTitles = true;
+		if ( reportLoaded ) {
+			const { startDate, endDate } = dateRangeDates;
+			const pageTitlesArgs = {
+				startDate,
+				endDate,
+			};
```

Comment:
We can use destruction here to make it a bit more concise:

```suggestion
			const pageTitlesArgs = {
				...dateRangeDates,
			};
```

Furthermore, we can re-struct the code in this section to avoid defining the `pageTitlesArgs` variable at all:

```js
( data.report?.[ 0 ]?.data?.rows || [] ).forEach(
    ...
);

data.titles = select( MODULES_ANALYTICS ).getPageTitles( {
    ...dateRangeDates,
    pagePaths,
} );
```

Reply:
`dateRangeDates` contains other properties that we don't need, so the best we can do is pull out the two args `startDate` `endDate` and omit creation of `pageTitlesArgs`

Reply:
https://github.com/google/site-kit-wp/pull/3958/commits/04f7489d6dead3d6306483f17adc53dfffd1e66e